scripts/init.sh

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
```

Comment:
@cpate4 so `init` and `utils` we are going to enforce as bash scripts (have extension), but the others not?

Reply:
Or is the idea that these 2 files are things that we are not expecting folks to need their own implementation of?

Do you desire people to add to their own version of `init.sh`, or should anything custom live in `setup`?

Reply:
... and to your last point, it is the that these 2 things we are not expecting folks to need their own implementations of. they are things the repo provided by way of help / organization - but the entry points are the standards. 

that being said, you are 100% correct. and using my own observations here, if it:
- isn't intuitive, and
- it isn't clear for the next person coming along, then
- it probably isn't a very good pattern

Reply:
@mflinn-broad / @pshapiro4broad - because you both weighted in before ... @cahrens is rightly pointing out these subtle nuances. thoughts? opinions? how do we make this clearer so people aren't sorting through a directory structure to figure out their entry points?

Reply:
... I am reviewing the PR one last time this morning, and in doing so, and in support of @cahrens's comments (again), I originally made `run-db` follow the `run` syntax, but with these additional scripts, should we?

in our world of black-and-white, we ran right into the gray 🤪

Reply:
My thinking with scripts like this is that no extension means it's an executable, .sh means it's a library that can be `source`'d into an executable. I realize that may not be apparent to everyone though. There isn't really a standard way to distinguish that type of thing with shell scripts that I'm aware of.

My opinion is that we should only be prescriptive about the `setup`, `build` and `run`. If teams want to add additional supporting logic, I think they should be encouraged to do so however works best for them, as long as core setup, build, and run interface is preserved.

I originally suggested the inclusion of utils.sh to @cpate4 because it contains a couple functions that are going to be generally useful in pretty much any script. ie a standardized way to get formatted log out put and to check that all the tools required to execute the script are installed on the user's machine.

Reply:
I don't have a strong opinion on the inclusion of an init.sh. I didn't include it in my implementation broadinstitute/yale because there just aren't really any environment variables to set there.

Reply:
I don't spend much time working with bash or python but I would lean toward always adding an extension, even for `run`. If we have a `scripts` gradle subproject, we can configure it to run `shellcheck` and `black` on PRs, which I think would require that they have correct extensions. (unless these tools look at all files for `#!` preambles?)

This how I set up the `scripts` subproject in `catalog`: https://github.com/DataBiosphere/terra-data-catalog/blob/main/scripts/build.gradle

Reply:
@pshapiro4broad - thanks for the feedback. [based on the referenced configs](https://github.com/DataBiosphere/terra-data-catalog/blob/main/scripts/build.gradle#L11) it is possible to configure gradle subprojects to include our scripts